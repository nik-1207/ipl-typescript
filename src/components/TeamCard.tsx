import { useCallback } from "react";
import { useHistory } from "react-router-dom";
import TeamCardStyle from "../styles/TeamCardStyle";
import TeamCardColour from "../styles/TeamCardColour";
interface TeamDataType {
  img: string;
  id: string;
  teamName: string;
  venue: string;
  winningYears: [number];
}
interface TeamCardStyleType {
  teamCard: string;
  image: string;
  tab: string;
  teamName: string;
  venue: string;
  win: string;
  teamHome: string;
  arrow: string;
}
interface TeamColourType {
  [key: string]: string;
}
function Card(props: TeamDataType) {
  const { img, id, teamName, venue, winningYears } = props;
  const history = useHistory<unknown>();
  const style: TeamCardStyleType = TeamCardStyle();
  const colour: TeamColourType = TeamCardColour();

  const clickHandler = useCallback(() => {
    history.push(`teams/${id}`);
  }, [id, history]);

  return (
    <div className={`${style.teamCard} ${colour[id]}`} onClick={clickHandler}>
      <img className={style.image} src={img} alt="team-logo" />
      <div className={style.tab}>
        <h2 className={style.teamName}>{teamName}</h2>
        <h3 className={style.venue}>{venue}</h3>
        {winningYears.length ? (
          <div className={style.win}>
            <i className="fas fa-trophy">
              {winningYears.map((item: number) => {
                return `  ${item}  `;
              })}
            </i>
          </div>
        ) : (
          <></>
        )}
      </div>
      <p className={style.teamHome}> Team Home</p>
      <i className={`fas fa-angle-right tab-arrow ${style.arrow}`} />
    </div>
  );
}

export default Card;
