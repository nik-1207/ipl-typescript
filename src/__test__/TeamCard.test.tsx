import { render, screen } from "@testing-library/react";
import TeamCard from "../components/TeamCard";

it("testing Player card", async () => {
  const id = "chennai-super-kings";
  const teamName = "Chennai Super Kings";
  const winningYears = [2010, 2011, 2018];
  const venue = "M. A. Chidambaram Stadium";

  render(
    <TeamCard
      teamName={teamName}
      img={""}
      id={id}
      venue={venue}
      winningYears={winningYears}
    />
  );
  const card = await screen.findByTestId("cards-teams");
  expect(card).toMatchSnapshot();
});
