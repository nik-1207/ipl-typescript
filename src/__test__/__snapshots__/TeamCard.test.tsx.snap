// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Player card 1`] = `
<div
  class="teamCard-0-2-1 chennai-super-kings-0-2-9"
  data-testid="cards-teams"
>
  <img
    alt="team-logo"
    class="image-0-2-2"
    src=""
  />
  <div
    class="tab-0-2-8"
  >
    <h2
      class="teamName-0-2-6"
    >
      Chennai Super Kings
    </h2>
    <h3
      class="venue-0-2-7"
    >
      M. A. Chidambaram Stadium
    </h3>
    <div
      class="win-0-2-4"
    >
      <i
        class="fas fa-trophy"
      >
          2010  
          2011  
          2018  
      </i>
    </div>
  </div>
  <p
    class="teamHome-0-2-5"
  >
     Team Home
  </p>
  <i
    class="fas fa-angle-right tab-arrow arrow-0-2-3"
  />
</div>
`;
